detector/src/main/java/com/synopsys/integration/detector/result/DetectorResult.java

```diff
@@ -42,4 +44,27 @@ public boolean getPassed() {
     public String getDescription() {
         return description;
     }
+
+    public String getStatusCode() {
+        if (!passed) {
+            return formatResultNameToStatusCode(resultName);
+        }
+        return "PASSED";
+    }
+
+    private String formatResultNameToStatusCode(String resultName) {
+        // Regular Expression
+        String regex = "([a-z])([A-Z]+)";
+
+        // Replacement string
+        String replacement = "$1_$2";
+
+        // Strip fully qualified class name
+        String[] classnamePieces = resultName.split("\\.");
+        String actualResultName = classnamePieces[classnamePieces.length-1].replace("DetectResult", "");
```

Comment:
Your example contains "DetectResult" but you try to strip it here?

Reply:
ah yeah my bad that was an example from before I added that replace, the strip works